java_tests/src/test/java/org/lsst/sal/SAL_Test_EvtLateJoinerOldestTest.java

```diff
@@ -0,0 +1,40 @@
+package org.lsst.sal;
+
+import Test.logevent_arrays;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SAL_Test_EvtLateJoinerOldestTest extends BaseTestCase {
+
+    /**
+     * Test that a late joiner can can read the most recent event using getNextSample.
+     * <p>
+     * Only one value should be retrievable but it turns out that all are. That is why
+     * this test case is in a separate file because otherwise all data from all previously
+     * run unit tests are retrievable. It is unclear why this is the case.
+     */
+    @Test
+    public void testEvtLateJoinerOldest() throws Exception {
+        final int numLoops = 5;
+        controller.salEventPub("Test_logevent_arrays");
+
+        logevent_arrays[] dataList = new logevent_arrays[numLoops];
+        for (int i = 0; i < numLoops; i++) {
+            logevent_arrays data = new logevent_arrays();
+            TestUtils.fillArraysWithRandomValues(data);
+            dataList[i] = data;
+            int retcode = controller.logEvent_arrays(data, 1);
+            Assert.assertEquals(retcode, SAL_Test.SAL__OK);
+        }
+
+        logevent_arrays data = new logevent_arrays();
+        remote.salEventSub("Test_logevent_arrays");
+
+        // In Python this next call gets the newest sample but in Java the oldest.
+        // This has been investigated but the cause for this is unclear.
```

Comment:
I would expect C++, Python, LabView and Java to all work the same (and C++ underlies Python and LabView, so I'd be amazed if any of those three differ from each other). I don't know which behavior @dmills-lsst considers correct. Please file a ticket to make them work the same as each other.

Reply:
@dmills-lsst has looked into this behavior and can't explain why this is different in Java. The expected behavior is the Python one.

Reply:
It turns out to be a bug in the SALPY unit test. I'll fix it using https://jira.lsstcorp.org/browse/DM-33212